java-squid/src/main/java/org/sonar/java/resolve/Types.java

```diff
@@ -131,27 +132,34 @@ private static Type best(List<Type> minimalCandidates) {
   }
 
   private static List<Set<Type>> supertypes(Iterable<Type> types) {
-    List<Set<Type>> results = new LinkedList<>();
-    for (Type type : types) {
-      checkParametrizedType(type);
-      Set<Type> supertypes = new LinkedHashSet<>();
-      supertypes.add(type);
-      for (Type supertype : ((JavaType) type).symbol.superTypes()) {
-        checkParametrizedType(supertype);
-        supertypes.add(supertype);
+    try {
+      List<Set<Type>> results = new LinkedList<>();
+      for (Type type : types) {
+        checkParametrizedType(type);
+        Set<Type> supertypes = new LinkedHashSet<>();
+        supertypes.add(type);
+        for (Type supertype : ((JavaType) type).symbol.superTypes()) {
+          checkParametrizedType(supertype);
+          supertypes.add(supertype);
+        }
+        results.add(supertypes);
       }
-      results.add(supertypes);
+      return results;
+    } catch (UnsupportedOperationException e) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1166)